src/systemctl/systemctl.c

```diff
@@ -4282,7 +4290,25 @@ static void print_status_info(
         else
                 printf("\n");
 
-        if (endswith(i->id, ".timer")) {
+        is_timer = endswith(i->id, ".timer");
+
+        if (!is_timer && !strv_isempty(i->triggered_by)) {
+                char **trigger;
+                UnitActiveState state;
```

Comment:
move this one to the inner loop? Also, please initialize to _UNIT_ACTIVE_STATE_INVALID, so that in the event get_state_one_unit() fails it's still properly initialized to something. And unit_active_state_to_string() then turns that to NULL, which al makes system.